Sources/UIAlertControllerView.swift

```diff
@@ -43,48 +57,103 @@ class UIAlertControllerView: UIView {
             return button
         }
 
+        clipsToBounds = true
         backgroundColor = .white
         layer.cornerRadius = 3
 
-        header.sizeToFit()
-        header.frame.height += verticalPadding
-        header.layer.contentsGravity = .top
         addSubview(header)
+        text.map { addSubview($0) }
+        buttons.forEach { addSubview($0) }
+
+        subviewWidth = {
+            let minWidth: CGFloat = 300
+            let maxWidth: CGFloat
+            if UIScreen.main.isPortrait {
+                maxWidth = UIScreen.main.bounds.width - 4 * horizontalPadding
+            } else {
+                maxWidth = max(minWidth, UIScreen.main.bounds.width / 1.5)
+            }
```

Comment:
I think in both these cases (but especially landscape) we should also apply a max readable width. For a font-size of 16px this is approximately 540px.

```suggestion
            let minWidth: CGFloat = 300
            let maxReadableWidth: CGFloat = 540
            let maxWidth: CGFloat
            if UIScreen.main.isPortrait {
                maxWidth = min(UIScreen.main.bounds.width - 4 * horizontalPadding, maxReadableWidth)
            } else {
                maxWidth = min(max(minWidth, UIScreen.main.bounds.width / 1.5), maxReadableWidth)
            }
```

The max readable width is important as when the text gets wider than that - it's very hard to read. I'll be presenting this in my Typography dev talk - hopefully in January.

Without max readable width
<img width="1392" alt="too-wide" src="https://user-images.githubusercontent.com/11625008/147130402-d18c26ec-d4e6-4082-8407-372425a24c4d.png">

After applying the max readable width
<img width="1392" alt="just-right" src="https://user-images.githubusercontent.com/11625008/147130393-13adfacc-fd3a-4886-8f06-fcf54132e377.png">